### PR TITLE
Third-party stubs: enforce CamelCase for type alias names

### PR DESCRIPTION
**Pull request by AlexWaygood**
Unblocks https://github.com/PyCQA/flake8-pyi/pull/243

### CODE REVIEW

File: stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi [hunk 1]
```diff
@@ -182,7 +182,7 @@ class WeakPopulateDict(dict[Any, Any]):
 
 column_set = set
 column_dict = dict
-ordered_column_set: TypeAlias = OrderedSet[ColumnElement[Any]]
+ordered_column_set: TypeAlias = OrderedSet[ColumnElement[Any]]  # noqa: Y042
```
Comment (AlexWaygood) [inline stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi:185]:
```suggestion
# Ignore Y042, this exists at runtime
ordered_column_set: TypeAlias = OrderedSet[ColumnElement[Any]]  # noqa: Y042
```

File: stubs/html5lib/html5lib/_tokenizer.pyi [hunk 2]
```diff
@@ -5,9 +5,9 @@ from typing_extensions import TypeAlias
 
 entitiesTrie: Any
 if sys.version_info >= (3, 7):
```
Comment (AlexWaygood) [inline stubs/html5lib/html5lib/_tokenizer.pyi:7]:
```suggestion
# ignore Y042, this exists at runtime
if sys.version_info >= (3, 7):
```

File: stubs/psutil/psutil/__init__.pyi [hunk 3]
```diff
@@ -109,8 +109,8 @@ elif sys.platform == "darwin":
 elif sys.platform == "win32":
     from ._pswindows import pfullmem, pmem
 else:
```
Comment (AlexWaygood) [inline stubs/psutil/psutil/__init__.pyi:111]:
```suggestion
else:
    # ignore Y042, these exist at runtime
```

**Comment by srittau** [comment]
A lot of aliases are marked as type aliases that I wouldn't consider type aliases. In my opinion as soon as an alias is used at runtime, I wouldn't consider it a type aliases. Examples are `ordered_column_set` in SQLAlchemy or `pmem` in psutil.

**Comment by AlexWaygood** [comment]
> A lot of aliases are marked as type aliases that I wouldn't consider type aliases. In my opinion as soon as an alias is used at runtime, I wouldn't consider it a type aliases. Examples are `ordered_column_set` in SQLAlchemy or `pmem` in psutil.

I agree that these don't _need_ to be type aliases necessarily. @hauntsaninja raised some objections the last time I tried to remove those markers from some of them (https://github.com/python/typeshed/pull/8248#pullrequestreview-1030484208), but I also don't think he has a _strong_ objection.

**Comment by AlexWaygood** [comment]
@srittau, does 27308f3 address your concerns?

File: stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi [hunk 4]
```diff
@@ -163,7 +161,11 @@ class WeakPopulateDict(dict[Any, Any]):
 
 column_set = set
 column_dict = dict
-ordered_column_set: TypeAlias = OrderedSet[ColumnElement[Any]]
+# Ignore Y026, this isn't a type alias.
+# We have to do `ordered_column_set = OrderedSet[_T]
+# instead of `ordered_column_set = OrderedSet`,
+# or pyright complains
```
Comment (JelleZijlstra) [inline stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi:167]: Why doesn't pyright complain about `column_set`?

File: stubs/psycopg2/psycopg2/_psycopg.pyi [hunk 5]
```diff
@@ -357,7 +357,7 @@ class Xid:
     def __getitem__(self, __index): ...
     def __len__(self): ...
 
-_T_cur = TypeVar("_T_cur", bound=_cursor)
+_T_cur = TypeVar("_T_cur", bound=_Cursor)
```
Comment (JelleZijlstra) [inline stubs/psycopg2/psycopg2/_psycopg.pyi:360]:
```suggestion
_T_cur = TypeVar("_T_cur", bound=cursor)
```
No need to use the alias here

**Review by JelleZijlstra — APPROVED** [review]
One optional suggestion

File: stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi (hunk 4 above)
Comment (AlexWaygood) [reply stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi:167]: I have no idea

**Pull request merged**